x-pack/plugins/security_solution/public/common/components/sourcerer/index.tsx

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useCallback, useMemo, useState } from 'react';
+import {
+  EuiButton,
+  EuiButtonEmpty,
+  EuiHighlight,
+  EuiIconTip,
+  EuiPopover,
+  EuiPopoverFooter,
+  EuiPopoverTitle,
+  EuiSelectable,
+} from '@elastic/eui';
+import { EuiSelectableOption } from '@elastic/eui/src/components/selectable/selectable_option';
+import { useManageSource } from '../../containers/sourcerer';
+import * as i18n from './translations';
+import { SOURCERER_FEATURE_FLAG_ON } from '../../containers/sourcerer/constants';
+import { ADD_INDEX_PATH } from '../../../../common/constants';
+
+export const MaybeSourcerer = React.memo(() => {
+  const {
+    activeSourceGroupId,
+    availableIndexPatterns,
+    getManageSourceGroupById,
+    isIndexPatternsLoading,
+    updateSourceGroupIndicies,
+  } = useManageSource();
+  const { defaultPatterns, indexPatterns: selectedOptions, loading: loadingIndices } = useMemo(
+    () => getManageSourceGroupById(activeSourceGroupId),
+    [getManageSourceGroupById, activeSourceGroupId]
+  );
+
+  const loading = useMemo(() => loadingIndices || isIndexPatternsLoading, [
+    isIndexPatternsLoading,
+    loadingIndices,
+  ]);
+
+  const onChangeIndexPattern = useCallback(
+    (newIndexPatterns: string[]) => {
+      updateSourceGroupIndicies(activeSourceGroupId, newIndexPatterns);
+    },
+    [activeSourceGroupId, updateSourceGroupIndicies]
+  );
+
+  const [isPopoverOpen, setPopoverIsOpen] = useState(false);
+  const trigger = useMemo(
+    () => (
+      <EuiButtonEmpty
+        aria-label={i18n.SOURCERER}
+        data-test-subj="sourcerer-trigger"
+        flush="left"
+        iconSide="right"
+        iconType="indexSettings"
+        onClick={() => setPopoverIsOpen(!isPopoverOpen)}
+        size="l"
+        title={i18n.SOURCERER}
+      >
+        {i18n.SOURCERER}
+      </EuiButtonEmpty>
+    ),
+    [isPopoverOpen]
+  );
+  const options: EuiSelectableOption[] = useMemo(
+    () =>
+      availableIndexPatterns.map((title, id) => ({
+        label: title,
+        key: `${title}-${id}`,
+        value: title,
+        checked: selectedOptions && selectedOptions.includes(title) ? 'on' : undefined,
```

Comment:
Given that `selectedOptions` is a string[], I think we can safely remove  `selectedOptions  &&`.
If something goes wrong after removing it, we might want to adjust the type of selectedOptions